File: PADTiltViewControllerExample/PADTiltViewControllerExample/PADCrawlDataSource.m
//
//  PADCrawlDataSource.m
//  PADTiltViewControllerExample
//
//  Created by Pavel Dusatko on 4/30/14.
//  Copyright (c) 2014 Pavel Dusatko. All rights reserved.
//

#import "PADCrawlDataSource.h"

static NSString *CellIdentifier = @"PADCrawlCell";

@interface PADCrawlDataSource ()

@property (nonatomic, strong) NSArray *array;

@end

@implementation PADCrawlDataSource

#pragma mark - Accessors

- (NSArray *)array {
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        NSString *path = [[NSBundle mainBundle] pathForResource:@"PADTiltViewControllerExample-Configuration" ofType:@"plist"];
        NSDictionary *dict = [NSDictionary dictionaryWithContentsOfFile:path];
        _array = dict[@"crawl"];
    });
    return _array;
}

#pragma mark - UITableViewDataSource

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    return self.array.count;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    id sectionInfo = self.array[section];
    return [sectionInfo count];
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:CellIdentifier forIndexPath:indexPath];
    cell.textLabel.text = self.array[indexPath.section][indexPath.row];
    return cell;
}

#pragma mark - UITableViewDelegate

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    [tableView deselectRowAtIndexPath:indexPath animated:YES];
}

@end

File: Swift-version/Inpiration/PADCrawlDataSource.swift
//
//  PADCrawlDataSource.swift
//  Inpiration
//
//  Created by Alexander Hüllmandel on 10/31/14.
//  Copyright (c) 2014 Alexander Hüllmandel. All rights reserved.
//

import Foundation
import UIKit


class PADCrawlDataSource: NSObject, UITableViewDelegate, UITableViewDataSource {
    let CellIdentifier = "PADCrawlCell"

    private lazy var array : Array<Array<String>> = {
        let path = NSBundle.mainBundle().pathForResource("PADTiltViewControllerExample-Configuration", ofType: "plist")
        let dict = NSDictionary(contentsOfFile: path!)
            return dict!["crawl"] as Array<Array<String>>
    }()
    
//MARK: - UITableViewDataSource
    
    func numberOfSectionsInTableView(tableView: UITableView) -> Int {
        return self.array.count
    }
    
    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        let sectionInfo = array[section] as Array
        return sectionInfo.count
    }
    
    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        var cell = tableView.dequeueReusableCellWithIdentifier(CellIdentifier, forIndexPath: indexPath) as UITableViewCell
        cell.textLabel.text = array[indexPath.section][indexPath.row]
        return cell
    }
    
//MARK: - UITableViewDelegate
    func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
        tableView.deselectRowAtIndexPath(indexPath, animated: true)
    }
}